js/adapt-stacklist.js

```diff
@@ -103,24 +101,36 @@ define(function(require) {
 
             if (isComplete) {
                 this.onComplete();
+                this.updateButton('', h);
+            } else {
+                var continueText = items[stage].next || this.model.get("_button").continueText;
+                var btnAriaLabel = this.model.get("_globals")._components._stacklist.ariaButtonLabel || this.model.get("_globals")._accessibility._ariaLabels.next;
+                var aria_label = continueText + ', ' + btnAriaLabel;
+
+                this.updateButton(continueText, h, aria_label);
             }
         },
 
-        updateButton: function(text, offset) {
+        updateButton: function(text, offset, aria_label) {
             this.$(".stacklist-button").css({ top: "+=" + offset });
 
-            var $button = this.$(".stacklist-next");
-            setTimeout(function() {
-                $button.html(text);
-            }, this.TRANSITION_TIME * 2);
+            if (text !== '') { // On last item we do not want to update text (it's most important when stack-list has only one item)
+                var $button = this.$(".stacklist-next");
+                $button.blur();
+                setTimeout(function() {
+                    $button.html(text);
+                    $button.attr('aria-label', aria_label);
+                }, this.TRANSITION_TIME * 2);
+            }
         },
 
         onComplete: function() {
-            var $button = this.$(".stacklist-button");
-            $button.css({ opacity: 0 });
+            var $button_div = this.$(".stacklist-button");
+            var $button = this.$(".stacklist-next");
+            $button_div.css({ opacity: 0 });
 
             setTimeout(function() {
-                $button.remove();
+                $button.attr('aria-hidden', 'true');
```

Comment:
Can I ask what the reason is for the changes in this method? Surely removing the element will make it unreadable just as setting `aria-hidden` will. Since once complete there's no further use for the button, I'd prefer it to be removed from the DOM.

Reply:
Originally when you click 'Next' button for last item, screen reader was not reading out last item text automatic, you would need to press up and down key arrows to read last item. I suspect that this was happening because of deleting element that screen reader is focused on. This was fixing the issue, however I've now double checked and appears I've also changed what `$button` is, and now `$button.remove()` only removes actual button, and leaves `.stacklist-button` div. In this way screen reader reads out last item automatic when it appears.